public/app/features/explore/ExploreToolbar.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import { hot } from 'react-hot-loader';
+
+import { ExploreId } from 'app/types/explore';
+import { DataSourceSelectItem, RawTimeRange, TimeRange } from '@grafana/ui';
+import { DataSourcePicker } from 'app/core/components/Select/DataSourcePicker';
+import { StoreState } from 'app/types/store';
+import { changeDatasource, clearQueries, splitClose, runQueries, splitOpen } from './state/actions';
+import TimePicker from './TimePicker';
+
+const createResponsiveButton = (options: {
+  splitted: boolean;
+  title: string;
+  onClick: () => void;
+  buttonClassName?: string;
+  iconClassName?: string;
+}) => {
+  const { title, onClick, buttonClassName, iconClassName, splitted } = options;
+
+  return (
+    <button className={`btn navbar-button ${buttonClassName ? buttonClassName : ''}`} onClick={onClick}>
+      <span className="btn-title">{!splitted ? title : ''}</span>
+      {iconClassName ? <i className={iconClassName} /> : null}
```

Comment:
I think the icons should be left, like they are on the timepicker. Having them right will look weird when the text goes away from the left when slimming the window width. Not sure what to do about the Run icon, perhaps it's worth making an exception here and have it right because it's at the right edge of the screen.

Reply:
The split icon is a bit too close to the text:
![screenshot 2019-01-28 at 10 35 41](https://user-images.githubusercontent.com/859729/51826938-8cb5ea00-22e8-11e9-8cf8-2c2b139cc705.png)

Reply:
Will fix